### ISSUE TITLE
Mix-up with release 0.27.0?

### ISSUE THREAD

**Issue by mkscrg** [facebook/flow#1979]
The 0.27.0 release notes mention

> - Fixed some issues with disjoint unions that have complex definitions

but they don't include links to issues PRs or commits. We assumed it referred to this union fix: 

> https://github.com/facebook/flow/commit/2df7671e7bda770b95e6b1eaede96d7a8ab1f2ac `fate of the union (a.k.a. the roads not taken)`

but it looks like 0.27.0 was cut from an earlier commit:

```
$ git log --graph --all '--pretty=%h [%ai]%d %s'
* 1cd8aa3 [2016-06-17 11:27:43 -0700] (HEAD -> master, origin/master, origin/HEAD) Fixes for node's 'net', 'child_process' lib
...
* 2df7671 [2016-06-10 00:30:07 -0700] fate of the union (a.k.a. the roads not taken)
...
* ab5c192 [2016-06-02 11:35:34 -0700] [hack] Editable syntax trees for the full-fidelity parser.
| * ceea4c3 [2016-06-10 15:41:54 -0700] (tag: v0.27.0, origin/v0.27) Bump version for 0.27.0
| * 96f6716 [2016-06-10 14:32:53 -0700] incremental fix: unchecked delete
| * ce9470b [2016-06-10 09:02:44 -0700] Revert "Handle JSX children"
| * 551604e [2016-06-09 05:25:31 -0700] fix tracking of const types after abnormal control flow
|/
* ccaf093 [2016-06-01 19:13:56 -0700] Remove unnecessary .bind in React docs
...
```

Also, the union fixes from https://github.com/facebook/flow/commit/2df7671e7bda770b95e6b1eaede96d7a8ab1f2ac seem to be deployed at tryflow.org, even though it says `v0.27.0`. [Example that doesn't actually type-check @ `v0.27.0`](https://tryflow.org/?code=Ly8gQGZsb3cKCmV4cG9ydCB0eXBlIEFjdGlvbiA9CiAgICB7IHR5cGU6ICdBJzsKICAgICAgc3RhdGU6IHsKICAgICAgICBpZDogc3RyaW5nOwogICAgICB9OwogICAgfQogIHwgeyB0eXBlOiAnQic7CiAgICAgIHN0YXRlOiB7CiAgICAgICAgaWQ6IHN0cmluZzsKICAgICAgICB1cGRhdGU6IHN0cmluZzsKICAgICAgfTsKICAgIH0KOwoKY29uc3QgbG9nID0gZnVuY3Rpb24gKGFjdGlvbjogQWN0aW9uKTogdm9pZCB7CiAgY29uc29sZS5sb2coYWN0aW9uKTsKfQoKY29uc3QgaWQgPSAndGVzdCc7CmNvbnN0IHVwZGF0ZSA9ICdmb29iYXInOwoKbG9nKHsKICB0eXBlOiAnQicsCiAgc3RhdGU6IHsgaWQsIHVwZGF0ZSB9LAp9KTs=)

Are the release notes wrong, or were some other union issues fixed in 0.27.0? And is tryflow.org not really running 0.27.0?

**Comment by jeffmo** [comment]
So the disjoint union fix that went out in 0.27 was [this one](https://github.com/facebook/flow/commit/a6e743dd76e303cd661102234c1f39e514db82f7).

The "fate of the union" commit is a much larger, much riskier change that we wanted to let sit for a bit and test, so it will be going out in 0.28 instead. It's the one that fixes tons of union issues (and also exposes lots of new errors).

As for tryflow.org, I'm not actually sure if that runs off of master or if it uses the latest tag? Either way it would say 0.27 until we've cut the 0.28 tag (the last step in that process for us is to commit the version number change as a marker for when the version goes out). (cc @ptmt)

**Comment by mkscrg** [comment]
Ok! Thanks for the context @jeffmo
